src/tir/usmp/analysis/extract_buffer_info.cc

```diff
@@ -491,6 +559,55 @@ BufferInfoAnalysis BufferInfoExtractor::operator()(const PrimFunc& main_func) {
       open_set.erase(le_event.buffer_info);
     }
   }
+
+  // All items in RO pool should have conflicts with each other in this RO pool
+  // as they will be placed in RO segment and pre-initialized
+  // Array<BufferInfo> buffer_info_arr =
+  //    ConvertToArrayOfBufferInfo(this->buffer_info_map_);
+
+  // split buffers to vars (RW) and constants (RO)
+  Array<BufferInfo> buffer_info_vars;
+  Array<BufferInfo> buffer_info_cons;
+  for (const auto& kv : this->buffer_info_map_) {
+    const auto& buf = kv.first;
+    for (const auto& pool : buf->pool_candidates) {
+      if (pool->IsInstance<ConstantPoolInfoNode>()) {
+        buffer_info_cons.push_back(buf);
+      } else {
+        buffer_info_vars.push_back(buf);
+      }
+
+      break;
+    }
+  }
+  ICHECK(buffer_info_map_.size() == buffer_info_vars.size() + buffer_info_cons.size())
+      << "missing value";
+
+  // intersect with each other, as all constants should exist at the same time
+  for (const auto& buf1 : buffer_info_cons) {
```

Comment:
We can avoid these expensive (with .find() on arrays) loop nests by simply replacing the conflicts array of each buffer node with a entire list of buffer nodes. Consider my other comment on line 598, I think it does not break to have all the buffer info objects to be conflicts of constant buffer info nodes (and that does make sense in theory as well).

Reply:
refactored